src/sst/core/event.cc

```diff
@@ -24,10 +24,68 @@ const SST::Event::id_type SST::Event::NO_ID = std::make_pair(0, -1);
 
 Event::~Event() {}
 
+extern std::map<std::string, uint64_t> eventHandlers;
+extern std::map<std::string, uint64_t> eventSendCounters;
+extern std::map<std::string, uint64_t> eventRecvCounters;
 
 void Event::execute(void)
 {
+
+    #ifdef EVENT_PROFILING
+    #ifdef HIGH_RESOLUTION_CLOCK
+    auto eventStart = std::chrono::high_resolution_clock::now();
+    #else
+    struct timeval eventStart, eventEnd, eventDiff;
+    gettimeofday(&eventStart, NULL);
+    #endif
+    #endif
+
     delivery_link->deliverEvent(this);
+
+    #ifdef EVENT_PROFILING
+    #ifdef HIGH_RESOLUTION_CLOCK
+    auto eventFinish = std::chrono::high_resolution_clock::now();
+    #else
+    gettimeofday(&eventEnd, NULL);
+    timersub(&eventEnd, &eventStart, &eventDiff);
+    #endif
+
+    // Track receiver processing time
+    auto eventHandler = eventHandlers.find(getLastComponentName());
+    if(eventHandler != eventHandlers.end())
+    {
+        #ifdef HIGH_RESOLUTION_CLOCK
+        eventHandler->second += std::chrono::duration_cast<std::chrono::nanoseconds>(eventFinish - eventStart).count();
+        #else
+        eventHandler->second += eventDiff.tv_usec + eventDiff.tv_sec * 1e6;
+        #endif
+    }
+
+    // Track sending and receiving counters
+    auto eventCount = eventRecvCounters.find(getLastComponentName());
```

Comment:
Probably better to use ComponentId's instead of names in this data structure.  The string can get rather long with SubComponents, and this will be a HUGE memory overhead if you're not careful.  You may need to dump the mapping to file somewhere to resolve things later.